### PR TITLE
Feat/update user seats

### PR DESCRIPTION
**Pull request by Ewan-Roberts**
## Changes

Closes [#1247(https://webiny.atlassian.net/browse/WEB-1247)

## How Has This Been Tested?

Manual

## Additional information
- This will fail until https://github.com/webiny/wcp/pull/194 is merged

### CODE REVIEW

File: packages/api-admin-users-cognito/src/createAdminUsers.ts [hunk 1]
```diff
@@ -94,34 +97,61 @@ export const createAdminUsers = ({
...
+            const wcpProjectEnvironment = getWcpProjectEnvironment();
+            const postUserSeatsEndpoint = getWcpApiUrl(
+                `/orgs/${wcpProjectEnvironment?.org.id}/projects/${wcpProjectEnvironment?.project.id}/package/seats`
+            );
+            if (wcpProjectEnvironment) {
+                await fetch(postUserSeatsEndpoint, {
+                    method: "POST",
+                    headers: { authorization: wcpProjectEnvironment.apiKey },
+                    body: JSON.stringify({ operation: "increment" })
                 });
```
Comment (Ewan-Roberts) [inline packages/api-admin-users-cognito/src/createAdminUsers.ts:110]: Im pretty sure I need to get the result from this and check the status code, if its failing I need to throw an error here

File: packages/api-wcp/src/createWcp.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ const cachedWcpProjectLicense: CachedWcpProjectLicense = {
 
 export const createWcp = async (): Promise<WcpContextObject> => {
     if (wcpProjectEnvironment) {
+        // TODO rename geWcpProjectLicenseCacheKey > getWcpProjectLicenseCacheKey
```
Comment (adrians5j) [inline packages/api-wcp/src/createWcp.ts:15]: Woah nice one. Will fix it on my side. Thanks!

File: packages/api-admin-users-cognito/package.json [hunk 3]
```diff
@@ -45,6 +46,7 @@
     "@webiny/handler-http": "^5.25.0",
     "@webiny/plugins": "^5.25.0",
     "@webiny/project-utils": "^5.28.0",
+    "node-fetch": "3.2.6",
```
Comment (adrians5j) [inline packages/api-admin-users-cognito/package.json:49]: Let's loose this one.

File: packages/api-admin-users-cognito/src/createAdminUsers.ts [hunk 4]
```diff
@@ -12,6 +12,9 @@ import { NotFoundError } from "@webiny/handler-graphql";
 import { AdminUser, AdminUsers, AdminUsersStorageOperations, CreatedBy, System } from "./types";
 import { createUserLoaders } from "./createAdminUsers/users.loaders";
 import { attachUserValidation } from "./createAdminUsers/users.validation";
+import { getWcpApiUrl } from "../../wcp/src/urls";
```
Comment (adrians5j) [inline packages/api-admin-users-cognito/src/createAdminUsers.ts:15]:
```suggestion
import { getWcpApiUrl } from "@webiny/wcp";
```

File: packages/api-admin-users-cognito/src/createAdminUsers.ts [hunk 5]
```diff
@@ -94,34 +97,61 @@ export const createAdminUsers = ({
...
-            delete (user as any)["password"];
-            try {
-                result = await storageOperations.createUser({ user });
-            } catch (err) {
-                throw WebinyError.from(err, {
-                    message: "Could not create user.",
-                    code: "CREATE_USER_ERROR",
-                    data: { user: result || user }
+
+            const wcpProjectEnvironment = getWcpProjectEnvironment();
```
Comment (adrians5j) [inline packages/api-admin-users-cognito/src/createAdminUsers.ts:101]: This wasn't the idea we discussed on the call.

We said we'll load the project environment outside of this file (upon calling `createAdminUsers` method) and we'll send the project environment to this `createAdminUsers` as an arg. This way we're not polluting / introducing a new dependency directly into this method, but we allow for this value to be passed outside of it. 

Let's do it like this.

1. Let's introduce a new `wcp: WcpContextObject` param for `createAdminUsers`
2. On the WcpContextObject side, let's add the incrementSeats / decrementSeats methods, which we'll then call within `createAdminUsers`. Much better than having all this WCP-specific code in a plugin whose purpose is something completely different.

**Review by adrians5j — CHANGES_REQUESTED** [review]
Please take a look at my comments.

Initially we discussed a different idea, so let's rearrange this a bit.

**Comment by adrians5j** [comment]
Closing this as I will be tackling it in a separate branch.

**Pull request closed without merging**